docs/running/server/wes.rst

```diff
@@ -0,0 +1,272 @@
+.. _workflowExecutionServiceOverview:
+
+Workflow Execution Service (WES)
+================================
+
+The GA4GH Workflow Execution Service (WES) is a standardized API for submitting and monitoring workflows.
+Toil has experimental support for setting up a WES server and executing CWL, WDL, and Toil workflows using the WES API.
+More information about the WES API specification can be found here_.
+
+.. _here: https://ga4gh.github.io/workflow-execution-service-schemas/docs/
+
+To get started with the Toil WES server, make sure that the ``server`` extra (:ref:`extras`) is installed.
+
+.. _WESUsageOverview:
+
+Starting a WES server
+---------------------
+
+To start a WES server on the default port 8080, run the Toil command::
+
+    $ toil server
+
+The WES API will be hosted on the following URL::
+
+    http://localhost:8080/ga4gh/wes/v1
+
+To use another port, e.g.: 3000, you can specify the ``--port`` argument::
+
+    $ toil server --port 3000
+
+There are many other command line options. Help information can be found by using this command::
+
+    $ toil server --help
+
+Below is a detailed summary of all available options:
+
+
+--debug
+            Enable debug mode.
+--host HOST
+            The host interface that the Toil server binds on. (default: "0.0.0.0").
+--port PORT
+            The port that the Toil server listens on. (default: 8080).
+--swagger_ui
+            Enable the swagger UI on the ``ga4gh/wes/v1/ui`` endpoint. (default: False).
+--cors
+            Enable Cross Origin Resource Sharing (CORS). This should only be turned on if the server is intended to be
+            used by a website or domain. (default: False).
+--cors_origins ORIGIN
+            Ignored if ``--cors`` is False. This sets the allowed origins for CORS. For details about CORS and its
+            security risks, see the `GA4GH docs on CORS`_. (default: "*").
+--workers WORKERS
+            Ignored if debug mode is on. The number of worker processes launched by the production WSGI server.
+            (default: 2).
+--opt ENGINE_OPTION
+            Specify the default parameters to be sent to the workflow engine for each run.  Accepts multiple values.
+
+            Example: ``toil server --opt=--logLevel=CRITICAL --opt=--workDir=/tmp``.
+
+.. _GA4GH docs on CORS: https://w3id.org/ga4gh/product-approval-support/cors
+
+
+Running on EC2
+--------------
+
+To run the server on a Toil leader instance on EC2:
+
+#. Launch a Toil cluster with the ``toil launch-cluster`` command with the AWS provisioner
+
+#. The Toil server requires an additional TCP port to open for HTTP(S) access. To add this inbound rule, open the AWS
```

Comment:
Is there a way to automate this step?

Reply:
I wouldn't recommend opening up this port to the Internet at all; I don't think the server has any auth. You probably want to access it over SSH forwarding for security; I think you can do something like `toil ssh-cluster -p aws -z us-west-2a --sshOption=-L8080:localhost:8080 cluster-name` to port forward it to your local machine.